dali/test/python/test_pipeline_debug.py

```diff
@@ -0,0 +1,144 @@
+        224, 224), mean=[0.485 * 255, 0.456 * 255, 0.406 * 255], std=[0.229 * 255, 0.224 * 255, 0.225 * 255])
+    return (output, labels.gpu())
+
+
+@pipeline_def(batch_size=8, num_threads=3, device_id=0)
+def load_images_pipeline():
+    jpegs, labels = fn.readers.file(
+        file_root=file_root, shard_id=0, num_shards=2)
+    images = fn.decoders.image(jpegs, output_type=types.RGB)
+    return images, labels
+
+
+@pipeline_def(batch_size=8, num_threads=3, device_id=0, debug=True)
+def numpy_array_injection_pipeline(images):
+    rng = fn.random.coin_flip(probability=0.5, seed=47)
+    images = fn.random_resized_crop(images, size=(224, 224), seed=27)
+    print(images.get().layout())
+    out_type = types.FLOAT16
+
+    output = fn.crop_mirror_normalize(images.gpu(), mirror=rng, device="gpu", dtype=out_type, crop=(
+        224, 224), mean=[0.485 * 255, 0.456 * 255, 0.406 * 255], std=[0.229 * 255, 0.224 * 255, 0.225 * 255])
+    return (output)
+
+
+def run_pipeline(func, *args, **kwargs):
+    pipe = func(*args, **kwargs)
+    pipe.build()
+    return pipe.run()
+
+
+def test_debug_pipeline_base():
+    rng, jpegs, labels, images, resized_images, output = run_pipeline(rn50_pipeline_base)
+    rng_, jpegs_, labels_, images_, resized_images_, output_ = run_pipeline(
+        rn50_pipeline_base, debug=True)
+
+    np.testing.assert_array_equal(rng.as_array(), rng_.as_array())
+    np.testing.assert_array_equal(rng.as_array(), rn50_pipeline_base_debug_values['rng'].as_array())
+    np.testing.assert_array_equal(labels.as_array(), labels_.as_array())
+    np.testing.assert_array_equal(labels.as_array(), labels_.as_array(),
+                                  rn50_pipeline_base_debug_values['labels'])
+    for j, j_, jd in zip(jpegs, jpegs_, rn50_pipeline_base_debug_values['jpegs']):
+        np.testing.assert_array_equal(j, j_)
+        np.testing.assert_array_equal(j, jd)
+    for i, i_, id in zip(images, images_, rn50_pipeline_base_debug_values['images']):
+        np.testing.assert_array_equal(i.as_cpu(), i_.as_cpu())
+        np.testing.assert_array_equal(i.as_cpu(), id.as_cpu())
+    for i, i_, id in zip(resized_images, resized_images_, rn50_pipeline_base_debug_values['resized_images']):
+        np.testing.assert_array_equal(i.as_cpu(), i_.as_cpu())
+        np.testing.assert_array_equal(i.as_cpu(), id.as_cpu())
+    for o, o_, od in zip(output, output_, rn50_pipeline_base_debug_values['output']):
+        np.testing.assert_array_equal(o.as_cpu(), o_.as_cpu())
+        np.testing.assert_array_equal(o.as_cpu(), od.as_cpu())
+
+
+def test_operations_on_debug_pipeline():
+    run_pipeline(rn50_pipeline)
+
+
+def test_numpy_injection():
+    load_pipeline = load_images_pipeline()
+    load_pipeline.build()
+    images_o, _ = load_pipeline.run()
+    images_o = [np.array(i) for i in images_o]
+    run_pipeline(numpy_array_injection_pipeline, images_o)
```

Comment:
General remark about the tests:
* We need a tests that runs more than one iteration and compares the result between debug and non-debug pipelines (see `compare_pipelines`).
* It might make sense, to do something similar as numpy_injection but it would be nice to check the result against something, for example a regular pipeline that loads and transforms the data.
* The `rn50_pipeline_base_debug_values` might be a list of values in case of running several iterations.
* We need some `external_source` testing as it has special handling.

In the long run (can be next PR), we may want a test similar to the cpu_only test, that verifies all the operators.

Reply:
done